channeldb/invoices.go

```diff
@@ -1166,6 +1194,8 @@ func deserializeInvoice(r io.Reader) (Invoice, error) {
 		// Invoice state.
 		tlv.MakePrimitiveRecord(invStateType, &state),
 		tlv.MakePrimitiveRecord(amtPaidType, &amtPaid),
+
+		tlv.MakePrimitiveRecord(hodlInvoiceType, &hodlInvoice),
```

Comment:
We could start to sort these after the fact, it's a bit easier on devs adding a new entry to this area.

Reply:
I think it is good as is. `NewStream` will check the canonical ordering and it saves an unnecessary sort call every time this is executed.